app/controllers/fileUpload/WhatYouWillNeedController.scala

```diff
@@ -48,29 +48,9 @@ class WhatYouWillNeedController @Inject()(
   def onPageLoad(srn: String, startDate: String, accessType: AccessType, version: Int, chargeType: ChargeType): Action[AnyContent] =
     (identify andThen getData(srn, startDate) andThen requireData andThen allowAccess(srn, startDate, None, version, accessType)).async { implicit request =>
       val ua = request.userAnswers
-
-      val psr = chargeType match {
-        case ChargeTypeLifetimeAllowance | ChargeTypeAnnualAllowance => ua.get(IsPublicServicePensionsRemedyPage(chargeType, optIndex = None))
-        case _ => None
-      }
-
-      val (templateDownloadLink, instructionsDownloadLink): (Json.JsValueWrapper, Json.JsValueWrapper) = (psr, chargeType) match {
-          case (Some(true), ChargeTypeAnnualAllowance) =>
-            (controllers.routes.FileDownloadController.templateFile(ChargeTypeAnnualAllowance, Some(true)).url,
-              controllers.routes.FileDownloadController.instructionsFile(ChargeTypeAnnualAllowance, Some(true)).url)
-          case (Some(true), ChargeTypeLifetimeAllowance) =>
-            (controllers.routes.FileDownloadController.templateFile(ChargeTypeLifetimeAllowance, Some(true)).url,
-              controllers.routes.FileDownloadController.instructionsFile(ChargeTypeLifetimeAllowance, Some(true)).url)
-          case (Some(false), ChargeTypeAnnualAllowance) =>
-            (controllers.routes.FileDownloadController.templateFile(ChargeTypeAnnualAllowance, Some(false)).url,
-              controllers.routes.FileDownloadController.instructionsFile(ChargeTypeAnnualAllowance, Some(false)).url)
-          case (Some(false), ChargeTypeLifetimeAllowance) =>
-            (controllers.routes.FileDownloadController.templateFile(ChargeTypeLifetimeAllowance, Some(false)).url,
-              controllers.routes.FileDownloadController.instructionsFile(ChargeTypeLifetimeAllowance, Some(false)).url)
-          case _ =>
-            (controllers.routes.FileDownloadController.templateFile(ChargeTypeOverseasTransfer, None).url,
-              controllers.routes.FileDownloadController.instructionsFile(ChargeTypeOverseasTransfer, None).url)
-        }
+      val (templateDownloadLink, instructionsDownloadLink) =
+        (controllers.routes.FileDownloadController.templateFile(chargeType, isPublicServicePensionsRemedy(chargeType,request)).url,
```

Comment:
We don't need to call the isPublicServicePensionsRemedy method twice. Store in val then only called once.